### PR TITLE
Update README to Pluggable Scanner Spec v1.1

### PR DESCRIPTION
**Pull request by prahaladdarkin**
*Scope of change*

* Update `README.md` to describe the new Pluggable Scanner Spec v1.1
* Update references to older Pluggable Scanner Spec v1.0 MIME types to the new v1.1 MIME type.
* Update request and response sample code snippets
* Update general guidance for scanners intending to implement the new specification

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -227,7 +257,7 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
 
 - The `Accept` request header is required to indicate to Scanner Adapter an intended scan report format
 - If the client does not specify the `Accept` header it's assumed to be Harbor vulnerability report with the
-  MIME type `application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0`.
+  MIME type `application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0`. This behavior is currently maintained only for backward compatibility purposes.
```
Comment (steven-zou) [inline README.md:261]: @prahaladdarkin 

Would you please also update the compatibility list doc link to the latest one `https://goharbor.io/docs/edge/install-config/harbor-compatibility-list/`?

Comment (steven-zou) [reply README.md:261]: And please also add the new scanner vendors to the list that is at the bottom of this doc.

[Sysdig](https://github.com/sysdiglabs/harbor-scanner-sysdig-secure)

[TensorSecurity](https://github.com/tensorsecurity/harbor-scanner)

**Review by steven-zou — CHANGES_REQUESTED** [review]
Please do changes.

**Comment by steven-zou** [comment]
Missing DCO?

add DCO by `git commit -s -m "my commit message"`

File: README.md [hunk 2]
```diff
@@ -136,51 +144,73 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
          "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
        }
        ```
-3. Try getting scan report (in unified Harbor format understandable by Harbor Web console):
+3. Try getting scan report (in the new v1.1 format understandable by Harbor Web console):
    ```
-   curl -H 'Accept: application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0' \
+   curl -H 'Accept: application/vnd.security.vulnerability.report; version=1.1' \
```
Comment (danielpacak) [inline README.md:148]: Let's add new MIME type of `application/vnd.security.vulnerability.report; version=1.1]` in 1. where we give example of metadata reply

File: README.md [hunk 3]
```diff
@@ -136,51 +144,73 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
...
+          },
+          "redhat": {
+            "V3Score": 7.5,
+            "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
+          }
+        }
+      }
+    }
+    ...
+  ]
```
Comment (danielpacak) [inline README.md]: Is this indentation okey?

File: README.md [hunk 4]
```diff
@@ -136,51 +144,73 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
...
-     "artifact": {
-       "repository": "library/mongo",
-       "digest": "sha256:917f5b7f4bef1b35ee90f03033f33a81002511c1e0767fd44276d4bd9cd2fa8e"
-     },
-     "scanner": {
-       "name": "Microscanner",
-       "vendor": "Aqua Security",
-       "version": "3.0.5",
-     },
-     "severity": "High",
```
Comment (danielpacak) [inline README.md:167]: Why are we deleting those properties? I think it's useful to keep them as reference

File: README.md [hunk 5]
```diff
@@ -136,51 +144,73 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
...
+      "fix_version": "1.1.1g-r0",
+      "severity": "High",
+      "description": "Server or client applications that call the SSL_check_chain() function during or after a TLS 1.3 handshake may crash due to a NULL pointer dereference as a result of incorrect handling of the \"signature_algorithms_cert\" TLS extension. The crash occurs if an invalid or unrecognised signature algorithm is received from the peer. This could be exploited by a malicious peer in a Denial of Service attack. OpenSSL version 1.1.1d, 1.1.1e, and 1.1.1f are affected by this issue. This issue did not affect OpenSSL versions prior to 1.1.1d. Fixed in OpenSSL 1.1.1g (Affected 1.1.1d-1.1.1f).",
+      "links": [
+        "https://avd.aquasec.com/nvd/cve-2020-1967"
+      ],
+      "layer": {
+        "digest": "sha256:9d48c3bd43c520dc2784e868a780e976b207cbf493eaff8c6596eb871cbd9609",
+        "diff_id": "sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0"
+      },
```
Comment (danielpacak) [inline README.md]: This is specific to Trivy. It's not specified by the API. Let's remove those extra properties

**Review by danielpacak — CHANGES_REQUESTED** [review]
👋 @prahaladdarkin Good changes. I left a few requests for you to take care of

File: README.md (hunk 1 above)
Comment (prahaladdarkin) [reply README.md:261]: Done. 
Just to confirm - would Sysdig and TensorSecurity confirm to the version 1.1 of the schema or the 1.0? If it is 1.0 I would  have to update the README.md on the 1.0 branch

File: README.md (hunk 2 above)
Comment (prahaladdarkin) [reply README.md:148]: Done.

File: README.md (hunk 5 above)
Comment (prahaladdarkin) [reply README.md]: Done. My bad - I copied the sample response snippet, but forgot to clean-up this property

File: README.md (hunk 3 above)
Comment (prahaladdarkin) [reply README.md]: I have increased the indentation. Hopefully things will look Ok now

**Review by prahaladdarkin — COMMENTED** [review]
Incorporated README.md related review comments

File: README.md (hunk 4 above)
Comment (prahaladdarkin) [reply README.md:167]: Done

File: README.md [hunk 6]
```diff
@@ -147,40 +156,76 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
...
+          "digest": "sha256:7215e8e09ea282e517aa350fc5380c1773c117b1867316fb59076d901e252d15",
+          "mime_type": "application/vnd.docker.distribution.manifest.v2+json"
+        },
+        "scanner": {
+          "name": "Trivy",
+          "vendor": "Aqua Security",
+          "version": "v0.16.0"
+        },
+        "severity": "High",
+      "vulnerabilities": [
```
Comment (danielpacak) [inline README.md]: Indentation: I believe the `vulnerabilities` property is at the same level as `severity`.

File: README.md [hunk 7]
```diff
@@ -147,40 +156,76 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
...
+                  "v2_vector": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
+                  "v3_vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
+                  "v2_score": 5,
+                  "v3_score": 7.5
+                },
+                "redhat": {
+                  "v3_vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
+                  "v3_score": 7.5
+                }
+              },
```
Comment (danielpacak) [inline README.md]: The `cvss` property is also specific to Trivy. We introduced it in the past for some other integration, but in API v1.1 it's redundant. See `.vendor_attributes.CVSS`.

**Review by danielpacak — CHANGES_REQUESTED** [review]
👋 @prahaladdarkin One more change request for the sample output returned by Trivy

**Review by prahaladdarkin — COMMENTED** [review]
@danielpacak I have incorporated the comments related to indentation and removal of the additional "cvss" attribute in the example snippet. I should have caught that  additional attribute much earlier. Thank you for pointing out.

**Review by steven-zou — COMMENTED** [review]
@prahaladdarkin 

Can we switch the default scanner and API sample with the default sample to `trivy` instead of `Clair`?

File: README.md [hunk 8]
```diff
@@ -147,40 +154,64 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
 4. Wait 15 seconds or use your own retry interval ...
 5. ... and try getting scan report again:
    ```
-   curl -H 'Accept: application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0' \
+    curl -H 'application/vnd.security.vulnerability.report; version=1.1' \
```
Comment (danielpacak) [inline README.md:157]:
```suggestion
    curl -H 'Accept: application/vnd.security.vulnerability.report; version=1.1' \
```

File: README.md [hunk 9]
```diff
@@ -63,9 +70,9 @@ Scanner vendors are supposed to implement Scanner Adapters exposing the Scanner
 
    {
      "scanner": {
-       "name": "Microscanner",
+       "name": "Trivy",
        "vendor": "Aqua Security",
-       "version": "3.0.5",
+       "version": "0.18.0",
```
Comment (danielpacak) [inline README.md:75]: To make it consistent with valid response:
```suggestion
       "version": "0.16.0",
```

**Review by danielpacak — APPROVED** [review]
Two small suggestions. Otherwise LGTM!

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**